### PR TITLE
CLI get and report results from racer 

### PR DESCRIPTION
**Pull request by JasonPapanicholasWF**
## Description
- Ping racer to start race
- Wait for results to be ready
- Fetch LHR and if specified HTML from the results endpoint
- Delete results
- Console log results
- Write to CSV and / or HTML 

## Type of Change

- [ ] Bug Fix
- [x] New Feature
- [ ] Breaking Change
- [ ] Refactor
- [ ] Documentation
- [ ] Other (please describe)

## Checklist

- [ ] I have read the [contributing guidelines](https://github.com/wayfair-incubator/oss-template/blob/main/CONTRIBUTING.md)
- [ ] Existing issues have been referenced (where applicable)
- [ ] I have verified this change is not present in other open pull requests
- [ ] Functionality is documented
- [ ] All code style checks pass
- [ ] New code contribution is covered by automated tests
- [ ] All new and existing tests pass

### CODE REVIEW

File: packages/race-cli/src/profile.ts [hunk 1]
```diff
@@ -39,45 +52,133 @@ export class ProfileScenario extends Scenario<ProfileContext> {
...
+        } else {
+          reject('Queue is not done processing');
+        }
+      });
+    };
+
     compose.buildAll({cwd: dockerPath}).then(() =>
-      compose.upAll({cwd: dockerPath, log: true}).then(
+      compose.upAll({cwd: dockerPath, log: isDebug}).then(
         async () => {
```
Comment (spember) [inline packages/race-cli/src/profile.ts]: I would feel better if this inner async function was broken up a bit, and did not mix await with Promise behavior

File: packages/proxy/src/proxy-worker.ts [hunk 2]
```diff
@@ -35,6 +36,23 @@ const handleProxyResponse = ({
...
+}: {
+  error: Error;
+  originalRequest: IncomingMessage;
+  responseToBrowser: ServerResponse | net.Socket;
+}) => {
+  if (responseToBrowser instanceof ServerResponse) {
+    responseToBrowser.writeHead(StatusCodes.NOT_FOUND);
+    responseToBrowser.end();
+  } else {
+    // console.log('What is a socket anyway?');
```
Comment (spember) [inline packages/proxy/src/proxy-worker.ts]: should this do a `console.error` instead?

File: packages/race-cli/src/reporters/console-reporter.ts [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+  initialize = (): Promise<void> => Promise.resolve();
+
+  process = (results: LighthouseResultsWrapper): Promise<void> =>
+    new Promise((resolve) => {
+      if (this._hasBegun === false) {
+        this._hasBegun = true;
+        console.log(this.buildHeaders());
+      }
+      console.log(this.buildRow(results.lhr));
```
Comment (spember) [inline packages/race-cli/src/reporters/console-reporter.ts:38]: I think we may have discussed this, but could we try a `console.table` instead?

File: packages/race-cli/src/scenarios/handlers.ts [hunk 4]
```diff
@@ -0,0 +1,175 @@
+import axios, {AxiosError, AxiosResponse} from 'axios';
+import logger from '../logger';
```
Comment (spember) [inline packages/race-cli/src/scenarios/handlers.ts:2]: oh good, common logger!

File: packages/race-cli/src/scenarios/handlers.ts [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+    .then((response: AxiosResponse) => {
+      if (response.status === StatusCodes.NO_CONTENT) {
+        logger.debug(`Success deleting ${jobId}`);
+      } else {
+        throw 'Bad response';
+      }
+    })
+    .catch((error: Error | AxiosError) => {
+      throw new Error(`Failed to delete ${jobId}`);
+      // Do some sort of cleanup here?
```
Comment (spember) [inline packages/race-cli/src/scenarios/handlers.ts]: I think maybe just a console.error is fine here. Currently it wouldn't be the end of the world if it failed to delete; the results will go away once the racing is done anyway

**Review by spember — APPROVED** [review]
I think this is a great start. a good base to build on for the future. 
Overall, I think a little more output regarding what's happening (e.g. a log message during container build... "building underlying containers" or something similar)

File: packages/race-cli/src/profile.ts (hunk 1 above)
Comment (JasonPapanicholasWF) [reply packages/race-cli/src/profile.ts]: Removed the then()s!

File: packages/race-cli/src/profile.ts [hunk 6]
```diff
@@ -39,45 +52,133 @@ export class ProfileScenario extends Scenario<ProfileContext> {
...
+    const checkQueue = async () => {
+      return new Promise<void>((resolve, reject) => {
+        if (numProcessed + numFailed === context.numberRuns) {
+          resolve();
+        } else {
+          reject('Queue is not done processing');
+        }
+      });
+    };
+
```
Comment (spember) [inline packages/race-cli/src/profile.ts]: right here would be a good spot for a log statement with something like "preparing docker images"

File: packages/race-cli/src/reporters/console-reporter.ts (hunk 3 above)
Comment (JasonPapanicholasWF) [reply packages/race-cli/src/reporters/console-reporter.ts:38]: Spoke offline, will handle in other ticket due to nature of console.table format

**Pull request merged**